ML project/Customer_churn_prediction.py

```diff
@@ -0,0 +1,512 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+
+# In[2]:
+
+
+df = pd.read_csv('train.csv')
+
+
+# In[3]:
+
+
+df.head()
+
+
+# In[4]:
+
+
+df.info()
+
+
+# In[5]:
+
+
+df.var()
+
+
+# Looks like 4 columns have high variance and would possibly benefit some kind of normalization
+
+# In[6]:
+
+
+sns.heatmap(df.corr(),cmap='BuGn');
+
+
+# Looks like there is multicollinearity among certain numerical features, hence some will be dropped
+
+# In[7]:
+
+
+df = df.copy()
+df.drop(['total_day_charge','total_eve_charge','total_night_charge','total_intl_charge'],axis=1,inplace=True)
+
+
+# In[8]:
+
+
+df.head()
+
+
+# In[9]:
+
+
+df.shape
+
+
+# In[10]:
+
+
+df[df.select_dtypes(['int64','float64']).columns].boxplot()
+plt.xticks(rotation=90);
+
+
+# The boxplot overview suggests that, since some outliers are in, the options are either proceed with the removal or adopt a robust scaling. Let's try out removal
+
+# In[11]:
+
+
+cols= list(df.select_dtypes(['int64','float64']).columns)
+
+upper_limit = df[cols].quantile(0.75) + 1.5*(df[cols].quantile(0.75)-df[cols].quantile(0.25))
+lower_limit = df[cols].quantile(0.25) - 1.5*(df[cols].quantile(0.75)-df[cols].quantile(0.25))
+cleaned_df = df[~((df[cols]>upper_limit)|(df[cols]<lower_limit)).any(axis=1)]
+cleaned_df.shape
+
+
+# In[12]:
+
+
+print('Percentage of removed data: {:.2f}'.format(100-cleaned_df.shape[0]/df.shape[0]*100))
+
+
+# In[13]:
+
+
+for i in cleaned_df.select_dtypes(['int64','float64']).columns:
+    sns.histplot(cleaned_df[i]);
+    plt.show();
+
+
+# The majority of numerical features shows a gaussian distribution, hence StandardScaler could be used too. Let's try MinMaxScaler
+
+# In[14]:
+
+
+from sklearn.preprocessing import MinMaxScaler
+scaler = MinMaxScaler()
+df_scaled = cleaned_df.copy()
+df_scaled[df_scaled.select_dtypes(['int64','float64']).columns] = scaler.fit_transform(df_scaled[df_scaled.select_dtypes(['int64','float64']).columns])
+
+
+# Let's format bynary categorical features and target column as needed
+
+# In[15]:
+
+
+df_scaled['international_plan'] = df_scaled.international_plan.apply(lambda x: 1 if x == 'yes' else 0)
+df_scaled['voice_mail_plan'] = df_scaled.voice_mail_plan.apply(lambda x: 1 if x == 'yes' else 0)
+df_scaled['churn'] = df_scaled.churn.apply(lambda x: 1 if x == 'yes' else 0)
```

Comment:
@Ettore1991 possible alternative is OrdinalEncoder https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OrdinalEncoder.html